x-pack/plugins/siem/public/components/page/network/domains_table/index.tsx

```diff
@@ -0,0 +1,224 @@
+  limit: number;
+}
+
+interface DomainsTableDispatchProps {
+  updateDomainsDirection: ActionCreator<{
+    flowDirection: FlowDirection;
+    networkType: networkModel.NetworkType;
+  }>;
+  updateDomainsLimit: ActionCreator<{
+    limit: number;
+    networkType: networkModel.NetworkType;
+  }>;
+  updateDomainsSort: ActionCreator<{
+    domainsSort: DomainsSortField;
+    networkType: networkModel.NetworkType;
+  }>;
+}
+
+type DomainsTableProps = OwnProps & DomainsTableReduxProps & DomainsTableDispatchProps;
+
+const rowItems: ItemsPerRow[] = [
+  {
+    text: i18n.ROWS_5,
+    numberOfRow: 5,
+  },
+  {
+    text: i18n.ROWS_10,
+    numberOfRow: 10,
+  },
+  {
+    text: i18n.ROWS_20,
+    numberOfRow: 20,
+  },
+  {
+    text: i18n.ROWS_50,
+    numberOfRow: 50,
+  },
+];
+
+const CountBadge = styled(EuiBadge)`
+  margin-left: 5px;
+`;
+
+export const DomainsTableId = 'domains-table';
+
+class DomainsTableComponent extends React.PureComponent<DomainsTableProps> {
+  public render() {
+    const {
+      data,
+      domainsSortField,
+      hasNextPage,
+      ip,
+      limit,
+      loading,
+      loadMore,
+      totalCount,
+      nextCursor,
+      updateDomainsLimit,
+      startDate,
+      endDate,
+      flowDirection,
+      flowTarget,
+      type,
+    } = this.props;
+
+    return (
+      <LoadMoreTable
+        columns={getDomainsColumns(
+          ip,
+          startDate,
+          endDate,
+          flowDirection,
+          flowTarget,
+          type,
+          DomainsTableId
+        )}
+        loadingTitle={i18n.DOMAINS}
+        loading={loading}
+        pageOfItems={data}
+        loadMore={() => loadMore(nextCursor)}
+        limit={limit}
+        hasNextPage={hasNextPage}
+        itemsPerRow={rowItems}
+        onChange={this.onChange}
+        sorting={getSortField(domainsSortField, flowTarget)}
+        updateLimitPagination={newLimit =>
+          updateDomainsLimit({ limit: newLimit, networkType: type })
+        }
+        title={
+          <EuiFlexGroup>
+            <EuiFlexItem>
+              <EuiFlexGroup>
+                <EuiFlexItem grow={false}>
+                  <h3>
+                    {i18n.DOMAINS}
+                    <CountBadge color="hollow">{totalCount}</CountBadge>
+                  </h3>
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <FlowDirectionSelect
+                id={DomainsTableId}
+                selectedDirection={flowDirection}
+                onChangeDirection={this.onChangeDomainsDirection}
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        }
+      />
+    );
+  }
+
+  private onChange = (criteria: Criteria) => {
+    if (criteria.sort != null) {
+      const splitField = criteria.sort.field.split('.');
+      const newDomainsSort: DomainsSortField = {
+        field: last(splitField) as DomainsFields,
```

Comment:
Nothing wrong if you want to use the `as` cast but you could also do function which does explicit checks and a default as well if it doesn't find it. The risk you run here is during runtime you really don't have a DomainsField.

Reply:
Added a function `getSortFromString()` for sorting out the `sort`'s type. 👍